### PR TITLE
feat: Cross Contract Interactions Chapter

### LINKED ISSUE #297: Building Advanced Contracts: Safe Dispatchers
https://github.com/starkware-libs/cairo/pull/3591/files

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #297

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/770)

### CODE REVIEW

**Review by enitrat — COMMENTED** [review]
Just finished a first read, i like it overall, will look for improvements later, but I can say for now that ch15-01 feels odd in its structure and the content 🤔

**Comment by TAdev0** [comment]
i personally dont like it much overall
i really dont like this chapter, how to originally present the thing

**Comment by enitrat** [comment]
Then rewrite until you have something you are satisfied with:)

**Comment by TAdev0** [comment]
Did you have ideas after first reading of what to improve ?

**Review by TAdev0 — COMMENTED** [review]
not much to say... its really better than mine. Very clear, the flow is great, i love it.

File: src/ch15-01-contract-class-abi.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+While this process is often abstracted by libraries and when using dispatchers, know that it's possible to call a function directly by providing its selector, for example when using a low-level system call like `starknet::call_contract_syscall` or when interacting with an RPC.
+
+## Encoding
+
+Smart contracts are written in a high-level languages like Cairo, using strong types to inform us about the data manipulated. However, the code executed on the blockchain is compiled into a sequence of low-level CASM instructions. The base data type in Starknet is `felt252`, and that's the only data manipulated at the CASM level. As such, all data must be serialized into `felt252` before being sent to the contract. The ABI specifies how types can be encoded into a sequence of `felt252`, and decoded back into their original form.
```
Comment (TAdev0) [inline src/ch15-01-contract-class-abi.md:42]:
```suggestion
Smart contracts are written in a high-level language like Cairo, using strong types to inform us about the data manipulated. However, the code executed on the blockchain is compiled into a sequence of low-level CASM instructions. The base data type in Starknet is `felt252`, and that's the only data manipulated at the CASM level. As such, all data must be serialized into `felt252` before being sent to the contract. The ABI specifies how types can be encoded into a sequence of `felt252`, and decoded back into their original form.
```

**Comment by TAdev0** [comment]
@enitrat  perfect! lets merge this!

File: src/ch15-03-executing-code-from-another-class.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+The call arguments must be provided as an array of arguments, serialized to a `Span<felt252>`. To serialize the arguments, we can simply use the `Serde` trait, provided that the types being serialized implement this trait. The call returns an array of serialized values, which we'll need to deserialize ourselves!
+
+## Summary
+
+Congratulations for finishing this chapter! You have learned a lot of new concepts:
+
+- How _Contracts_ differ from _Classes_ and how the ABI describes them for external sources.
+- How to call functions from other contracts and classes using the _Dispatcher_ pattern.
+- How to use _Library calls_ to execute the logic of another class in the context of the caller.
+- The two syscalls that Starknet provides to interact with contracts and classes:
```
Comment (TAdev0) [inline src/ch15-03-executing-code-from-another-class.md:70]:
```suggestion
- The two syscalls that Starknet provides to interact with contracts and classes.
```

**Comment by enitrat** [comment]
@TAdev0 thanks, but please don't merge suggestions yourself :)

**Comment by TAdev0** [comment]
oh ok, i should have pushed a normal commit from my local branch instead? doing this was quicker, my bad

**Comment by enitrat** [comment]
no i mean usually don't commit the suggestions yourself

**Pull request merged**